src/java.base/share/classes/java/lang/Object.java

```diff
@@ -33,6 +33,25 @@
  * Every class has {@code Object} as a superclass. All objects,
  * including arrays, implement the methods of this class.
  *
+ * @apiNote
+ * <h2><a id=objectOverrides>Guidance for overriding {@code Object}
+ * methods</a></h2>
+ *
+ * {@code java.lang.Object} includes three {@code public} methods
+ * overridable in subclasses, {@link #equals(Object)}, {@link
+ * #hashCode()}, and {@link #toString()}. Each of these methods can be
+ * called by any code that has access to an object of a
+ * class. Especially if an object can be used by code in a separate
+ * maintenance domain, any overrides of these methods should take care
+ * to provide reasonable implementations of these methods. In
+ * particular, the implementations should take care to avoid using
+ * excessive memory, computational time, or any other resources.
+ * Additionally, during typical operation these methods should
+ * <em>not</em> throw any exception or other throwable; as always, a
+ * {@link VirtualMachineError} is possible during the execution of a
+ * method, often due to factors outside of the method's direct
+ * control.
```

Comment:
"Should not throw any exception or other throwable" is overly broad. However, there is a narrower sense where code that implements these methods "shouldn't" throw anything. I'd suggest focusing on precondition checking. Specifically, no object should ever be in a state such that calling one of these methods results in IllegalStateException or other exception based on the state of the object. In addition, no argument passed to equals() should ever cause IllegalArgumentException, ClassCastException, NullPointerException, or other exception based on the argument.

Reply:
I would hope to spend as little space on this as possible, perhaps "This method should avoid throwing or propagating any exceptions unless it legitimately _cannot_ adhere to this contract."

Reply:
> I would hope to spend as little space on this as possible, perhaps "This method should avoid throwing or propagating any exceptions unless it legitimately _cannot_ adhere to this contract." (or shorter)

Pushed a version using that worked, but I expect discussion will continue.

Reply:
Since you mentioned "normally" from the JLS, can we something like:

> during typical operation these methods should complete normally, that is, they do not throw any exception, except unrecoverable ones outside of the method's direct control, like {@link VirtualMachineError}.

(The term "exception" seems to refer to general Throwable in JLS)